rustls/src/server/hs.rs

```diff
@@ -202,7 +202,7 @@ impl ExtensionProcessing {
                 if let Some(resume) = resumedata {
                     if sess.config.max_early_data_size > 0
                         && hello.early_data_extension_offered()
-                        && resume.version == sess.common.negotiated_version.unwrap()
+                        && Some(resume.version) == sess.common.negotiated_version
```

Comment:
Can we split the server side into a separate PR?

I think we shouldn't be using `sess.common.negotiated_version` here. Instead, should pass a non-`Option` copy of the negotiated version so that it is clear that version negotiation definitely happened before this point; i.e. `sess.common.negotiated` version is never `None` here.

(I think the `unwrap()` was actually better than this, because that made it clearer.)

Reply:
Reverted. I will look at a better server side implementation elsewhere.